signature/signature-v4.go

```diff
@@ -186,7 +203,14 @@ func V4SignVerify(r *http.Request) ErrorCode {
 
 	// Verify if signature match.
 	if !compareSignatureV4(newSignature, signV4Values.Signature) {
-		return errSignatureDoesNotMatch
+		//try unsigned payload
```

Comment:
Can't we detect the user is using "UNSIGNED-PAYLOAD" by looking in `X-Amz-Content-Sha256`? Then we don't need to guess .

Reply:
Thanks for the suggestion. That sounds reasonable, I'll do some further tests.